src/expr/src/scalar/func.rs

```diff
@@ -2052,6 +2052,31 @@ fn parse_ident<'a>(
     })?)
 }
 
+fn regexp_split_to_array<'a>(
+    text: Datum<'a>,
+    regexp: Datum<'a>,
+    flags: Datum<'a>,
+    temp_storage: &'a RowArena,
+) -> Result<Datum<'a>, EvalError> {
+    // TODO: Figure out if this remakes regexes each call and avoid doing that.
```

Comment:
Could a compute person comment on this?

Reply:
I think this is true, we don't have a way to cache compiled regexes here.

Reply:
https://github.com/MaterializeInc/materialize/blob/537da883b38a113eb9312bc0acb3d768874773dd/src/expr/src/scalar/mod.rs#L1114 appears to do this for another regex function, because there's other internal func variants that it can call. Would it make sense to have (in a followup PR) another internal func that can have this cached regex so we could optimize it similarly?

Reply:
Yep, if the inputs are literal, then we should evaluate it as early as possible. For non-literal arguments, we have to rebuild the regex compiler on every invocation unfortunately.

Reply:
> Yep, if the inputs are literal, then we should evaluate it as early as possible.

Yes. [This](https://github.com/MaterializeInc/materialize/blob/14cb1e26774a80d876b9deec9044f0e89d778a31/src/expr/src/scalar/mod.rs#L780-L793) is the code that does this for `IsRegexpMatch`, by transforming it to the unary version, which has the regex hardcoded.

Reply:
And in the long term, we should figure out how to also handle it efficiently when the regex is not a literal. I created an issue: https://github.com/MaterializeInc/materialize/issues/20592